src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -1981,7 +1981,7 @@ static void window_ride_construction_entrance_click(rct_window* w)
     else
     {
         gRideEntranceExitPlaceType = ENTRANCE_TYPE_RIDE_ENTRANCE;
-        gRideEntranceExitPlaceRideIndex = w->number & 0xFF;
+        gRideEntranceExitPlaceRideIndex = w->number % MAX_RIDES;
```

Comment:
This doesn't make sense. The original code was being silly and using & 0xFF instead of a type cast. When this window becomes a class, I would expect a field like _rideIndex or something to be used.

Reply:
I wasn't sure about that, so I figured I'd play it safe until the window gets refactored.

Reply:
Yeah, it would just cause confusion if it did wrap.